ww3tools/wread.py

```diff
@@ -1619,3 +1619,224 @@ def spec_ww3(*args):
+                            dire[k] = float(line[i])*180/pi
+                            k = k+1
+
+                    nl = int(floor((nf*nd)/7.))
+                    rnl = int(np.round(7*((float(nf*nd)/7)-nl)))
+                    auxs = np.zeros((nf*nd), 'f')
+                    wnds = np.zeros((nt), 'f')
+                    wndd = np.zeros((nt), 'f')
+
+                    for t in range(0, nt):
+
+                        cabc = fp.readline()
+                        cabc.strip().split()[0]
+                        mtime[t] = np.double(timegm(strptime(
+                            cabc.strip().split()[0]+cabc.strip().split()[1][0:2], '%Y%m%d%H')))
+                        cabc = fp.readline()
+                        cabc = cabc.strip().split()
+                        
+                        if len(cabc) >8:
+                            
+                            namep = cabc[0][1:]
+                            lat = float(cabc[2])
+                            lon = float(cabc[3])
+                            depth = float(cabc[4])
+                            wnds_index = 5
+                            wndd_index = 6
+                        elif len(cabc) == 8:
+                            
+                            namep = cabc[0][1:]
+                            lat_lon_str = cabc[2]
+                            lat_lon_str = lat_lon_str.strip("'")
+                            lat_lon_parts = lat_lon_str.split('-')
+                            lat = float(lat_lon_parts[0])
+                            lon = -float(lat_lon_parts[1])
+
+                            depth = float(cabc[3])
+                            wnds_index = 4
+                            wndd_index = 5
+                        else:
+                            continue 
+                            
+
+                        wnds[t] = float(cabc[wnds_index])
+                        wndd[t] = float(cabc[wndd_index])
+
+                        k = 0
+                        for i in range(0, nl):
+                            line = fp.readline()
+                            line = line.strip().split()
+                            for j in range(0, 7):
+                                auxs[k] = float(line[j])
+                                k = k+1
+
+                        if rncd > 0:
+                            line = fp.readline()
+                            line = line.strip().split()
+                            for i in range(0, rnl):
+                                auxs[k] = float(line[i])
+                                k = k+1
+
+                        for ic in range(0, nf):
+                            for il in range(0, nd):
+                                dspec[t, ic, il] = auxs[il*nf+ic]
+
+                    fp.close()
+                    del fp
+
+                    mdate = pd.to_datetime(mtime, unit='s').strftime(
+                        '%Y-%m-%dT%H:%M:%S.%f')
+                    freq1 = freq*np.nan
```

Comment:
You can see the function spec_ww3 in
https://github.com/NOAA-EMC/WW3-tools/blob/develop/ww3tools/wread.py
If it is a spectrum generated by ww3 in netcdf format, it will contain freq1 and freq2. From the text spec point output we only have one "freq" array.
For the same function spec_ww3 you will see the power spectrum, pwst . And this 4.01*np.sqrt(sum(pwst[t,:])) should give you the same (or very close) to Hs. 

Having the freq1 and freq2 is very nice because it allows spectral interpolation. See the "Spectral Interpolation Block" in
https://github.com/NOAA-EMC/WW3-tools/blob/develop/examples/comparevalww3board.py
I want to move this block to become a function when I have time.